http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -238,18 +222,17 @@ class LeoPubsubMessageSubscriber[F[_]](
       .parEvalMapUnordered(config.concurrency)(messageHandler)
       .handleErrorWith(error => Stream.eval(logger.error(error)("Failed to initialize message processor")))
 
-  private def ack(event: CloudPubsubEvent[LeoPubsubMessage]): F[Unit] =
+  private def ack(event: ReceivedMessage[LeoPubsubMessage]): F[Unit] =
     for {
       _ <- logger.info(s"acking message: ${event}")
-      _ <- event match {
-        case CloudPubsubEvent.GCP(event) =>
-          F.delay(
-            event.consumer.ack()
-          )
-        case CloudPubsubEvent.Azure(_) => F.unit
-      }
+
+      _ <- F.delay(event.ackHandler.ack())
+
       end <- F.realTimeInstant
-      duration = (end.toEpochMilli - event.publishedTime.toEpochMilli).millis
+      // A zero duration means the publishing time is not available, this should never be case
+      // and it must be debugged if it happens.
+      duration = Duration.create( end.toEpochMilli - event.publishedTime.getOrElse(Instant.now()).toEpochMilli,
```

Comment:
this is interesting ... curiosity thought / questions.

were you experiencing an empty `.publishedTime`?, is this something that is specific to Azure Pub/Sub?

... and to your comment, I would like to lean towards at least logging this with a `.WARN`, or marking it as expected behavior because as is this will silently produce a duration of 0 (but, at the same time, this is actually an expected condition, so I am kind of wondering if anything really needs to be debugged)